cpp/oneapi/dal/algo/pca/backend/gpu/train_kernel_svd_impl_dpc.cpp

```diff
@@ -0,0 +1,207 @@
+auto svd_decomposition(sycl::queue& q,
+                       pr::ndview<Float, 2>& data,
+                       std::size_t row_count,
+                       std::size_t column_count) {
+    auto U = pr::ndarray<Float, 2>::empty({ row_count, row_count });
+    auto S = pr::ndarray<Float, 1>::empty({ column_count });
+    auto V_T = pr::ndarray<Float, 2>::empty({ column_count, column_count });
+
+    //auto result = mkl::lapack::gesvd(queue, uplo, trans, n, k, alpha, a, lda, beta, c, ldc, deps);
+    Float* data_ptr = data.get_mutable_data();
+    Float* U_ptr = U.get_mutable_data();
+    Float* S_ptr = S.get_mutable_data();
+    Float* V_T_ptr = V_T.get_mutable_data();
+
+    const std::size_t min_dim = std::min(row_count, column_count);
+    const std::size_t max_dim = std::max(row_count, column_count);
+    const std::size_t num_iters = 100;
+
+    q.submit([&](sycl::handler& cgh) {
+        auto data_acc = sycl::accessor(data_ptr, cgh, sycl::read_write);
+        auto U_acc = sycl::accessor(U_ptr, cgh, sycl::write);
+        auto S_acc = sycl::accessor(S_ptr, cgh, sycl::write);
+        auto V_T_acc = sycl::accessor(V_T_ptr, cgh, sycl::write);
+
+        cgh.parallel_for(sycl::range<1>(1), [=](sycl::id<1> idx) {
+            for (std::size_t iter = 0; iter < num_iters; ++iter) {
+                matrix_multiply(data_acc.get_pointer(),
+                                data_acc.get_pointer(),
+                                U_acc.get_pointer(),
+                                row_count,
+                                column_count,
+                                row_count,
+                                column_count,
+                                row_count,
+                                row_count);
+
+                singular_value_decomposition(U_acc.get_pointer(),
+                                             min_dim,
+                                             max_dim,
+                                             U_acc.get_pointer(),
+                                             S_acc.get_pointer(),
+                                             V_T_acc.get_pointer());
+
+                matrix_multiply(U_acc.get_pointer(),
+                                S_acc.get_pointer(),
+                                V_T_acc.get_pointer(),
+                                row_count,
+                                min_dim,
+                                min_dim,
+                                max_dim,
+                                row_count,
+                                column_count);
+            }
+        });
+    });
+
+    q.wait();
+
+    return std::make_tuple(U, S, V_T);
+}
+
+template <typename Float>
+result_t train_kernel_svd_impl<Float>::operator()(const descriptor_t& desc, const input_t& input) {
+    ONEDAL_ASSERT(input.get_data().has_data());
+    const auto data = input.get_data();
+
+    ONEDAL_ASSERT(data.get_column_count() > 0);
+    std::int64_t column_count = data.get_column_count();
```

Comment:
Please make it const:
```suggestion
    const std::int64_t column_count = data.get_column_count();
```